Cython/Utility/ImportExport.c

```diff
@@ -329,17 +330,20 @@ static int ${import_star}(PyObject* m) {
     PyObject *utf8_name = 0;
     PyObject *name;
     PyObject *item;
+    PyObject *import_obj;
 
     locals = PyDict_New();              if (!locals) goto bad;
     if (__Pyx_import_all_from(locals, m) < 0) goto bad;
     list = PyDict_Items(locals);        if (!list) goto bad;
 
-    for(i=0; i<PyList_GET_SIZE(list); i++) {
-        name = PyTuple_GET_ITEM(PyList_GET_ITEM(list, i), 0);
-        item = PyTuple_GET_ITEM(PyList_GET_ITEM(list, i), 1);
+    for(i=0; i<PyList_Size(list); i++) {
```

Comment:
`__Pyx_PyList_GET_SIZE` makes this selection at compile-time. In principle it can also fail so if probably wants a check `#if !CYTHON_ASSUME_SAFE_SIZE`.

It's a bit debatable how much we really have to have a check given the only real failure is "not a list" and `PyDict_Items` is specified as returning a list. I can't quite remember what we decided on unfortunately.